JASP-Engine/JASP/R/descriptives.R

```diff
@@ -1145,4 +1186,90 @@ Descriptives <- function(jaspResults, dataset, options) {
   return(SEK)
 }
 
-# </editor-fold> HELPER FUNCTIONS BLOCK
\ No newline at end of file
+.descriptivesQQPlot <- function(dataset, options,  qqvar, levelName=NULL) {
+
+  if (options$descriptivesQQPlot && is.null(options$variables))
+    return(list(result=NULL))
+
+  descriptivesQQPlot <- list()
+
+  if (!is.null(qqvar)) {
+
+    #to put a subtitle if there is a split
+    if(!is.null(levelName))
+      descriptivesQQPlot$title <- paste0(options$splitby, ": ", levelName)
+    else
+      descriptivesQQPlot$title <- " "
+
+    # Hardcode plot dimensions
+    options$descriptivesPlotWidthQQPlot <- 400
+    options$descriptivesPlotHeightQQPlot <- 400
+
+    descriptivesQQPlot$width  <- options$descriptivesPlotWidthQQPlot
+    descriptivesQQPlot$height <- options$descriptivesPlotHeightQQPlot
+
+    varCol<-dataset[[.v(qqvar)]]
+    varCol<-varCol[!is.na(varCol)]
+
+    standResid <- as.data.frame(stats::qqnorm(varCol, plot.it=FALSE))
+
+    standResid <- na.omit(standResid)
+    xVar <- standResid$x
+    yVar <- standResid$y
+    yVar<-yVar-mean(yVar)
+    yVar<-yVar/(sd(yVar))
+
+    # Format x ticks
+    xlow   <- min(pretty(xVar))
+    xhigh  <- max(pretty(xVar))
+    xticks <- pretty(c(xlow, xhigh))
+
+    # format x labels
+    xLabs <- vector("character", length(xticks))
+    for (i in seq_along(xticks)) {
+      if (xticks[i] < 10^6) {
+        xLabs[i] <- format(xticks[i], digits= 3, scientific = FALSE)
+      } else {
+        xLabs[i] <- format(xticks[i], digits= 3, scientific = TRUE)
+      }
+    }
+
+    # Format y ticks
+    ylow   <- min(pretty(yVar))
+    yhigh  <- max(pretty(yVar))
+    yticks <- pretty(c(ylow, yhigh))
+
+    # format y labels
+    yLabs <- vector("character", length(yticks))
+    for (i in seq_along(yticks)) {
+      if (yticks[i] < 10^6) {
+        yLabs[i] <- format(yticks[i], digits= 3, scientific = FALSE)
+      } else {
+        yLabs[i] <- format(yticks[i], digits= 3, scientific = TRUE)
+      }
+    }
+    p <- JASPgraphs::drawAxis(xName = "Theoretical Quantiles", yName = paste0("Standardised Residuals"), xBreaks = xticks, yBreaks = xticks, yLabels = xLabs, xLabels = xLabs, force = TRUE)
+    p <- p + ggplot2::geom_line(data = data.frame(x = c(min(xticks), max(xticks)), y = c(min(xticks), max(xticks))), mapping = ggplot2::aes(x = x, y = y), col = "darkred", size = 1)
+    p <- JASPgraphs::drawPoints(p, dat = data.frame(xVar, yVar), size = 3)
+
+    # JASP theme
+    p <- JASPgraphs::themeJasp(p)
+
+  } else {
+
+    descriptivesQQPlot$title <- "Q-Q Plot"
+
+    # Hardcode plot dimensions
+    options$descriptivesPlotWidthQQPlot  <- 400
+    options$descriptivesPlotHeightQQPlot <- 400
+
+    descriptivesQQPlot$width  <- options$descriptivesPlotWidthQQPlot
+    descriptivesQQPlot$height <- options$descriptivesPlotHeightQQPlot
+    descriptivesQQPlot$custom <- list(width="descriptivesPlotWidthQQPlot", height="descriptivesPlotHeightQQPlot")
+    descriptivesQQPlot$data   <- NULL
```

Comment:
Don't need the four lines above here; this is sort of a mix of old-style plotting and new-style plotting.
I totally understand that it is a little confusing that we have two distinct styles of plotting in JASP right now.